### PR TITLE
Support External Help Link

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**
If there is an external "help" link supplied in the config, the "help" option inside the account dropdown will direct users to the external web address. An icon will also be added indicating the link is an external link.

image

**PR Checklist:**
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

### CODE REVIEW

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM

File: lib/util/auth.js [hunk 1]
```diff
@@ -5,19 +5,24 @@ import { isOtpMiddleware } from './user'
...
-  },
-  {
-    messageId: 'help',
-    // Add a target attribute if you need the link to open in a new window, etc.
-    // (supports the same values as <a target=... >).
-    // target: '_blank',
-    url: '/help'
-  }
-]
+export const accountLinks = (config) => {
```
Comment (miles-grant-ibigroup) [inline lib/util/auth.js:8]: nitpick: can we refactor this to take in `extraMenuItems` instead of the entire config?

File: lib/util/auth.js [hunk 2]
```diff
@@ -5,19 +5,24 @@ import { isOtpMiddleware } from './user'
...
+  const helpLink =
+    config?.extraMenuItems.find((link) => link.id === 'help') || undefined
+  const helpLinkUrl = helpLink ? helpLink.href : '/help'
```
Comment (miles-grant-ibigroup) [inline lib/util/auth.js:11]: Can we inline this?

File: lib/util/auth.js [hunk 3]
```diff
@@ -5,19 +5,24 @@ import { isOtpMiddleware } from './user'
...
+  return [
+    {
+      messageId: 'myAccount',
+      url: ACCOUNT_PATH
+    },
+    {
+      messageId: 'help',
+      // Add a target attribute if you need the link to open in a new window, etc.
+      // (supports the same values as <a target=... >).
+      // target: '_blank',
```
Comment (miles-grant-ibigroup) [inline lib/util/auth.js:21]: Is this supposed to be uncommented?

File: lib/components/mobile/navigation-bar.js [hunk 4]
```diff
@@ -110,6 +112,7 @@ class MobileNavigationBar extends Component {
 const mapStateToProps = (state) => {
   return {
     auth0Config: getAuth0Config(state.otp.config.persistence),
+    config: state.otp.config,
```
Comment (miles-grant-ibigroup) [inline lib/components/mobile/navigation-bar.js:115]: Once accountLinks is extracted this can be refactored to only extract the part of the config we need

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Works! Just a few nitpicks

File: lib/util/auth.js (hunk 3 above)
Comment (amy-corson-ibigroup) [reply lib/util/auth.js:21]: This exists in dev, but I think it's just a demonstration? Seems like weird behavior for this to open a new tab by default

Comment (miles-grant-ibigroup) [reply lib/util/auth.js:21]: Good call. We can leave it commented out for now!

**Pull request merged**